Storage/src/Connection/Rest.php

```diff
@@ -44,8 +44,13 @@ class Rest implements ConnectionInterface
     use UriTrait;
 
     const BASE_URI = 'https://storage.googleapis.com/storage/v1/';
```

Comment:
I believe we should deprecate this constant and add a new one (replacing all references to `BASE_URI`) called `DEFAULT_API_ENDPOINT` set the `https://storage.googleapis.com`.